v3/control.go

```diff
@@ -651,18 +629,52 @@ func encodeControls(controls []Control) *ber.Packet {
 
 // ControlDirSync implements the control described in https://msdn.microsoft.com/en-us/library/aa366978(v=vs.85).aspx
 type ControlDirSync struct {
-	Flags      int64
-	MaxAttrCnt int64
-	Cookie     []byte
+	Criticality  bool
+	Flags        int64
+	MaxAttrCount int64
+	Cookie       []byte
 }
 
-// NewControlDirSync returns a dir sync control
-func NewControlDirSync(flags int64, maxAttrCount int64, cookie []byte) *ControlDirSync {
+// NewRequestControlDirSync returns a dir sync control
+func NewRequestControlDirSync(
```

Comment:
I found a good name in RFC.

>    Controls sent by clients are termed 'request controls', and those sent by servers are termed 'response controls'.
> 
> https://datatracker.ietf.org/doc/html/rfc4511#section-4.1.11

Reply:
Unfortunately, the function has been exposed in the latest release and is therefore considered immutable. I suggest an alias instead with the same function signature and add a mark to the current one to mark it as deprecated.